### ISSUE TITLE
Upgrade Operator SDK Dependency

### ISSUE THREAD

**Issue by jmckind** [argoproj-labs/argocd-operator#49]
Currently the operator depends on v0.12.0 of the operator-sdk. The operator needs to be updated to use the current version of the operator-sdk to prevent getting too far behind, resulting in a more challenging upgrade in the future.

**Comment by jmckind** [comment]
Merged into master and will be released in v0.0.9
